silo-core/contracts/incentives/base/DistributionManager.sol

```diff
@@ -0,0 +1,318 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity 0.8.28;
+
+import {IERC20} from "openzeppelin5/token/ERC20/IERC20.sol";
+
+import {Ownable2Step, Ownable} from "openzeppelin5/access/Ownable2Step.sol";
+import {EnumerableSet} from "openzeppelin5/utils/structs/EnumerableSet.sol";
+
+import {IDistributionManager} from "../interfaces/IDistributionManager.sol";
+import {DistributionTypes} from "../lib/DistributionTypes.sol";
+import {TokenHelper} from "../../lib/TokenHelper.sol";
+
+/**
+ * @title DistributionManager
+ * @notice Accounting contract to manage multiple staking distributions
+ */
+contract DistributionManager is IDistributionManager, Ownable2Step {
+    using EnumerableSet for EnumerableSet.Bytes32Set;
+
+    EnumerableSet.Bytes32Set internal _incentivesProgramIds;
```

Comment:
We should have a getter to read whole array.

Reply:
Done. See `getAllProgramsNames()`.